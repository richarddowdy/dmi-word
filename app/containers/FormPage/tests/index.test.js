/**
 *
 * Tests for Form
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import configureStore from '../../../configureStore';


import { FormPage } from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

describe('<Form />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });
  
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <FormPage dispatch={dispatch} />
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  // it('Expect to have additional unit tests specified', () => {
  //   expect(true).toEqual(false);
  // });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <FormPage />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
