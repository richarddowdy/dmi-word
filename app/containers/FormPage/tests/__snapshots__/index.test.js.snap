// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> Should render and match the snapshot 1`] = `
<div>
  <form
    class="Form-sc-197cms6-0 dAgUQH"
  >
    <label
      for="newString"
    >
      Add a New Word or Phrase to the list!
    </label>
    <input
      class="form-group Input-sc-16ykmnb-0 joywpR"
      id="new-string"
      placeholder="New Phrase!"
      type="text"
      value=""
    />
    <button
      class="btn btn-primary"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
